### ISSUE TITLE
"make clean test" fails on  "String.length(build_info[:revision]) == 7"

### ISSUE THREAD

**Issue by s4uliu5** [elixir-lang/elixir#5580]
### Environment

* Elixir & Erlang versions (elixir -v):

```
Erlang/OTP 19 [erts-8.1.1] [source] [64-bit] [smp:4:4] [async-threads:10] [hipe] [kernel-poll:false]
Elixir 1.3.4
```

* Operating system:
  Ubuntu 14.04
```
Linux e64 4.4.0-57-generic #78~14.04.1-Ubuntu SMP Sat Dec 10 00:14:47 UTC 2016 x86_64 x86_64 x86_64 GNU/Linux
```

### Current behavior

Executing 
```sh
git clone https://github.com/elixir-lang/elixir.git
cd elixir
make clean test
```
one test fails

```
==> elixir (eunit)
  All 194 tests passed.

==> elixir (exunit)
warning: variable user is unused
  test/elixir/kernel/fn_test.exs:24

Excluding tags: [otp19: false]
..........................................................................................................
  1) test build_info/0 (SystemTest)
     test/elixir/system_test.exs:7
     Assertion with == failed
     code: String.length(build_info[:revision]) == 7
     lhs:  9
     rhs:  7
     stacktrace:
       test/elixir/system_test.exs:16: (test)
........................................................................................................................
Finished in 14.9 seconds (8.1s on load, 6.8s on tests)
3133 tests, 1 failure, 1 skipped

Randomized with seed 64297
make: *** [test_stdlib] Error 1
```

### Expected behavior

All tests to pass.

**Comment by josevalim** [comment]
This has been fixed on bfffa9430b135388717ce59326082aa6feb36775, which is on master. When did you clone Elixir and which branch are you running on?

**Comment by povilas** [comment]
@josevalim this commit was not merged to v1.3 branch
https://github.com/elixir-lang/elixir/blob/v1.3/lib/elixir/test/elixir/system_test.exs#L16
and also missing in v1.4 
https://github.com/elixir-lang/elixir/blob/v1.4/lib/elixir/test/elixir/system_test.exs#L16

**Comment by s4uliu5** [comment]
I cloned an hour before opened this issue.
To be exact I performed these steps

```
git clone https://github.com/elixir-lang/elixir.git
cd elixir
git checkout v1.3.4
make clean test
```

I missed to note that I checkout'ed version 1.3.4, before running make.

**Comment by s4uliu5** [comment]
I just locally applied the bfffa94 and the original issue is gone, but got another failure

```
Generated iex app
==> elixir (eunit)
  All 194 tests passed.

==> elixir (exunit)
warning: variable user is unused
  test/elixir/kernel/fn_test.exs:24

Excluding tags: [otp19: false]

....................................................... .......

Finished in 15.0 seconds (8.1s on load, 6.8s on tests)
3133 tests, 0 failures, 1 skipped

Randomized with seed 312094
==> ex_unit (exunit)
...................................................................................................................................................

  1) test it supports timeouts (ExUnitTest)
     test/ex_unit_test.exs:47
     Assertion with =~ failed
     code: output =~ ~r"\\(stdlib\\) timer\\.erl:\\d+: :timer\\.sleep/1"
     lhs:  "\n\n  1) test ok (ExUnitTest.TimeoutTest)\n     \e[1m\e[30mtest/ex_unit_test.exs:52\e[0m\n     \e[31m** (ExUnit.TimeoutError) test timed out after10ms. You can change the timeout:\n     \n       1. per test by setting \"@tag timeout: x\"\n       2. per case by setting \"@moduletag timeout: x\"\n       3. globally via \"ExUnit.start(timeout: x)\" configuration\n       4. or set it to infinity per run by calling \"mix test --trace\"\n          (useful when using IEx.pry)\n     \n     Timeouts are given as integers in milliseconds.\n     \e[0m\n     \e[36mstacktrace:\e[0m\n       test/ex_unit_test.exs:53: (test)\n  (ex_unit) lib/ex_unit/runner.ex:296: ExUnit.Runner.exec_test/1\n       (ex_unit) lib/ex_unit/runner.ex:246: anonymous fn/3 in ExUnit.Runner.spawn_test/3\n\n\n\nFinished in 0.01 seconds\n\e[31m1 test, 1 failure\e[0m\n\nRandomized with seed 739315\n"
     rhs:  ~r/\(stdlib\) timer\.erl:\d+: :timer\.sleep\/1/
     stacktrace:
       test/ex_unit_test.exs:61: (test)

..............................................................

Finished in 1.7 seconds (0.8s on load, 0.9s on tests)
210 tests, 1 failure

Randomized with seed 337742
make: *** [test_ex_unit] Error 1
```

Should I close this issue, as solution is clear, and open another with the new failure?

**Comment by josevalim** [comment]
Do you get that failure in a second run?

Seems to be a race conditions rather.

**Comment by josevalim** [comment]
Backported the fix to 1.4 as well.

**Comment by s4uliu5** [comment]
Great job!

Just pulled "mater" branch and ```make clean test``` finished without any problem.
